Not specifying theme in new environment causes invalid memory address or nil pointer dereference

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

Hello,

I tried to install the latest oh-my-posh (26.14.0) on a host and I got the error:

oh-my-posh panic: runtime error: invalid memory address or nil pointer dereference

When executing:

eval "$(oh-my-posh init bash)"

This persisted, until I copied over 26.8.0 from another machine (trying to download that specific version using the install.sh script failed), ran it once, then 26.14.0 started to work.

I created a new user on my local WSL system to see if I could replicate the error and it happened there too, so I am able to provide you with a stack trace:

```
$ eval "$(oh-my-posh init bash)"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xb066f3]

goroutine 1 [running]:
github.com/jandedobbeleer/oh-my-posh/src/config.(*Config).upgradeFeatures(0xc0001e26e0, {0xf82638, 0xc0000a3020})
        D:/a/oh-my-posh/oh-my-posh/src/config/config.go:193 +0xb3
github.com/jandedobbeleer/oh-my-posh/src/config.(*Config).Features(0xc0001e26e0, {0xf82638, 0xc0000a3020})
        D:/a/oh-my-posh/oh-my-posh/src/config/config.go:134 +0x3d9
github.com/jandedobbeleer/oh-my-posh/src/cli.runInit({0x7fffaff275df, 0x4})
        D:/a/oh-my-posh/oh-my-posh/src/cli/init.go:98 +0x234
github.com/jandedobbeleer/oh-my-posh/src/cli.createInitCmd.func1(0xc0001c8900?, {0xc000043290?, 0x4?, 0xd8a17a?})
        D:/a/oh-my-posh/oh-my-posh/src/cli/init.go:54 +0x2f
github.com/spf13/cobra.(*Command).execute(0xc0002ca308, {0xc000043260, 0x1, 0x1})
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1019 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0x15b6640)
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x46f
github.com/spf13/cobra.(*Command).Execute(...)
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/jandedobbeleer/oh-my-posh/src/cli.Execute()
        D:/a/oh-my-posh/oh-my-posh/src/cli/root.go:94 +0x1a
main.main()
        D:/a/oh-my-posh/oh-my-posh/src/main.go:8 +0xf
```

I then tested running 26.14.0 in this clean environment with a theme specified on the command line and then it worked without error after.

Then I re-created the clean user and home directory so I could get you debugging output.

### Theme

Didn't specify a theme, that appears to be the issue.

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

bash

### Log output

```Shell
Version: 26.14.0

Shell: bash

Prompt:

 omp-test   ~  CONFIG ERROR                                                                   in bash at 23:47:14

Segments:

ConsoleTitle(true)                        -   0 ms
Session(true)                             -   2 ms
Path(true)                                -   0 ms
Git(false)                                -   0 ms
Root(false)                               -   0 ms
Status(true)                              -   0 ms
Node(false)                               -   0 ms
Go(false)                                 -   0 ms
Python(false)                             -   0 ms
Shell(true)                               -   0 ms
Time(true)                                -   0 ms

Run duration: 13.386067ms

Cache path: /home/omp-test/.cache/oh-my-posh

Config path: no --config set, using default built-in configuration

Logs:

[DEBUG] 23:47:14.837 debug.go:35 → logging enabled, raw mode: true
[TRACE] 23:47:14.837 load.go:filePath() - 871ns
[ERROR] 23:47:14.837 load.go:Load:37 → no config file specified
[TRACE] 23:47:14.837 load.go:Load() - 43.92µs
[DEBUG] 23:47:14.837 home.go:12 → /home/omp-test
[TRACE] 23:47:14.837 path.go:Path() - 36.382µs
[DEBUG] 23:47:14.837 file.go:Init:26 → loading cache file: /home/omp-test/.cache/oh-my-posh/omp.cache
[DEBUG] 23:47:14.838 file.go:Init:47 → loading cache key: environment_platform
[DEBUG] 23:47:14.838 file.go:Init:47 → loading cache key: is_wsl
[TRACE] 23:47:14.838 file.go:Init(/home/omp-test/.cache/oh-my-posh/omp.cache) - 114.953µs
[DEBUG] 23:47:14.838 file.go:Get:69 → cache is nil, returning empty value for key: prompt_count_cache
[TRACE] 23:47:14.838 terminal.go:setPromptCount() - 21.238µs
[TRACE] 23:47:14.838 terminal.go:GOOS() - 33ns
[DEBUG] 23:47:14.838 terminal.go:setPwd:115 → /home/omp-test
[TRACE] 23:47:14.838 terminal.go:setPwd() - 13.876µs
[TRACE] 23:47:14.838 terminal.go:Init() - 211.081µs
[DEBUG] 23:47:14.838 terminal.go:Shell:368 → no shell name provided in flags, trying to detect it
[DEBUG] 23:47:14.839 terminal.go:Shell:379 → process name: bash
[TRACE] 23:47:14.839 terminal.go:Shell() - 1.717276ms
[TRACE] 23:47:14.839 terminal.go:Flags() - 22ns
[DEBUG] 23:47:14.839 file.go:Get:69 → cache is nil, returning empty value for key: template_cache
[TRACE] 23:47:14.839 cache.go:restoreCache() - 3.064µs
[TRACE] 23:47:14.839 terminal_unix.go:Root() - 408ns
[TRACE] 23:47:14.839 terminal.go:Shell() - 77ns
[TRACE] 23:47:14.839 terminal.go:Flags() - 22ns
[TRACE] 23:47:14.839 terminal.go:StatusCodes() - 33ns
[DEBUG] 23:47:14.839 file.go:Get:80 → getting cache key: is_wsl with value: true
[DEBUG] 23:47:14.839 terminal_unix.go:IsWsl:32 → true
[TRACE] 23:47:14.839 terminal_unix.go:IsWsl() - 5.324µs
[TRACE] 23:47:14.839 terminal.go:Flags() - 23ns
[TRACE] 23:47:14.839 terminal.go:Flags() - 22ns
[DEBUG] 23:47:14.839 home.go:12 → /home/omp-test
[DEBUG] 23:47:14.839 file.go:Get:80 → getting cache key: is_wsl with value: true
[DEBUG] 23:47:14.839 terminal_unix.go:IsWsl:32 → true
[TRACE] 23:47:14.839 terminal_unix.go:IsWsl() - 25.448µs
[DEBUG] 23:47:14.843 terminal.go:RunCommand:293 → //wsl.localhost/Ubuntu/home/omp-test
[TRACE] 23:47:14.843 terminal.go:RunCommand(wslpath -m /home/omp-test) - 3.733186ms
[TRACE] 23:47:14.843 terminal.go:Flags() - 231ns
[TRACE] 23:47:14.843 terminal.go:Flags() - 132ns
[TRACE] 23:47:14.843 terminal.go:GOOS() - 33ns
[DEBUG] 23:47:14.843 terminal.go:User:249 → omp-test
[TRACE] 23:47:14.843 terminal.go:User() - 6.536µs
[DEBUG] 23:47:14.843 terminal.go:Host:266 → Mini-Marvel
[TRACE] 23:47:14.843 terminal.go:Host() - 6.249µs
[TRACE] 23:47:14.843 terminal.go:GOOS() - 22ns
[DEBUG] 23:47:14.843 file.go:Get:80 → getting cache key: environment_platform with value: ubuntu
[DEBUG] 23:47:14.843 terminal_unix.go:Platform:92 → ubuntu
[DEBUG] 23:47:14.843 terminal.go:Getenv:77 → 1
[TRACE] 23:47:14.843 terminal.go:Getenv(SHLVL) - 4.551µs
[DEBUG] 23:47:14.843 writer.go:Init:106 → terminal program: Unknown
[DEBUG] 23:47:14.843 writer.go:Init:107 → terminal shell: shell
[DEBUG] 23:47:14.843 terminal.go:Getenv:77 → NO DATA
[TRACE] 23:47:14.843 terminal.go:Getenv(OMP_CACHE_DISABLED) - 4.761µs
[DEBUG] 23:47:14.843 file.go:Get:69 → cache is nil, returning empty value for key: accent_color
[TRACE] 23:47:14.843 colors.go:SetAccentColor() - 5.753µs
[TRACE] 23:47:14.843 terminal.go:Shell() - 209ns
[DEBUG] 23:47:14.843 terminal.go:Getenv:77 → NO DATA
[TRACE] 23:47:14.843 terminal.go:Getenv(POSH_SHELL_VERSION) - 2.469µs
[DEBUG] 23:47:14.843 debug.go:PrintDebug:24 → segment: Title
[DEBUG] 23:47:14.843 text.go:patchTemplate:125 → {{ .Shell }} in {{ .Folder }}
[TRACE] 23:47:14.844 text.go:Render({{ .Shell }} in {{ .Folder }}) - 300.331µs
[TRACE] 23:47:14.844 terminal.go:Flags() - 22ns
[DEBUG] 23:47:14.844 terminal.go:Getenv:77 → NO DATA
[TRACE] 23:47:14.844 terminal.go:Getenv(POSH_CURSOR_LINE) - 3.296µs
[DEBUG] 23:47:14.844 terminal.go:Getenv:77 → NO DATA
[TRACE] 23:47:14.844 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 6.833µs
[TRACE] 23:47:14.844 terminal.go:Flags() - 243ns
[TRACE] 23:47:14.844 terminal.go:Flags() - 22ns
[TRACE] 23:47:14.844 terminal.go:Flags() - 77ns
[DEBUG] 23:47:14.844 home.go:12 → /home/omp-test
[TRACE] 23:47:14.844 terminal.go:GOOS() - 34ns
[TRACE] 23:47:14.844 terminal.go:Flags() - 440ns
[DEBUG] 23:47:14.844 segment.go:Execute:113 → segment: Status
[DEBUG] 23:47:14.844 file.go:Get:69 → cache is nil, returning empty value for key: toggle_cache
[DEBUG] 23:47:14.844 segment.go:isToggled:234 → no toggles found
[TRACE] 23:47:14.844 terminal.go:StatusCodes() - 44ns
[DEBUG] 23:47:14.844 properties.go:GetString:28 → status_template: {{ .Code }}
[TRACE] 23:47:14.844 terminal.go:Flags() - 352ns
[TRACE] 23:47:14.844 terminal.go:Flags() - 221ns
[DEBUG] 23:47:14.844 home.go:12 → /home/omp-test
[TRACE] 23:47:14.844 terminal.go:GOOS() - 121ns
[DEBUG] 23:47:14.844 segment.go:Execute:113 → segment: Path
[DEBUG] 23:47:14.844 home.go:12 → /home/omp-test
[TRACE] 23:47:14.844 terminal.go:GOOS() - 33ns
[DEBUG] 23:47:14.844 segment.go:Execute:113 → segment: Root
[DEBUG] 23:47:14.844 file.go:Get:69 → cache is nil, returning empty value for key: toggle_cache
[DEBUG] 23:47:14.844 file.go:Get:69 → cache is nil, returning empty value for key: toggle_cache
[TRACE] 23:47:14.844 terminal_unix.go:Root() - 628ns
[DEBUG] 23:47:14.844 segment.go:isToggled:234 → no toggles found
[DEBUG] 23:47:14.844 properties.go:GetBool:22 → display_cygpath: false
[TRACE] 23:47:14.844 terminal.go:GOOS() - 44ns
[TRACE] 23:47:14.844 separator.go:Separator() - 55ns
[TRACE] 23:47:14.844 terminal.go:Shell() - 88ns
[TRACE] 23:47:14.844 terminal.go:Shell() - 44ns
[TRACE] 23:47:14.844 terminal.go:GOOS() - 33ns
[DEBUG] 23:47:14.844 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 23:47:14.844 properties.go:GetString:28 → windows_registry_icon: 
[DEBUG] 23:47:14.844 home.go:12 → /home/omp-test
[TRACE] 23:47:14.844 terminal.go:GOOS() - 77ns
[DEBUG] 23:47:14.844 properties.go:GetString:28 → windows_registry_icon: 
[DEBUG] 23:47:14.844 segment.go:Execute:113 → segment: Git
[DEBUG] 23:47:14.844 home.go:12 → /home/omp-test
[TRACE] 23:47:14.844 separator.go:Separator() - 88ns
[DEBUG] 23:47:14.844 file.go:Get:69 → cache is nil, returning empty value for key: toggle_cache
[DEBUG] 23:47:14.844 segment.go:isToggled:234 → no toggles found
[DEBUG] 23:47:14.844 text.go:patchTemplate:125 → {{ .Code }}
[TRACE] 23:47:14.844 text.go:Render({{ .Code }}) - 341.066µs
[DEBUG] 23:47:14.844 terminal.go:ResolveSymlink:208 → /home/omp-test
[TRACE] 23:47:14.844 terminal.go:ResolveSymlink(/home/omp-test) - 21.954µs
[DEBUG] 23:47:14.844 properties.go:GetBool:22 → always_enabled: true
[TRACE] 23:47:14.844 terminal.go:Flags() - 342ns
[DEBUG] 23:47:14.844 home.go:12 → /home/omp-test
[TRACE] 23:47:14.844 terminal.go:GOOS() - 55ns
[DEBUG] 23:47:14.844 segment.go:Execute:113 → segment: Session
[TRACE] 23:47:14.844 terminal.go:GOOS() - 66ns
[TRACE] 23:47:14.844 terminal.go:GOOS() - 23ns
[ERROR] 23:47:14.844 terminal.go:HasParentFilePath:477 → stat .git: no such file or directory
[TRACE] 23:47:14.844 terminal.go:HasParentFilePath(.git) - 78.781µs
[DEBUG] 23:47:14.844 properties.go:GetString:28 → home_icon: ~
[DEBUG] 23:47:14.844 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 23:47:14.844 separator.go:Separator() - 22ns
[TRACE] 23:47:14.844 terminal.go:GOOS() - 22ns
[TRACE] 23:47:14.844 terminal.go:GOOS() - 22ns
[TRACE] 23:47:14.844 separator.go:Separator() - 33ns
[DEBUG] 23:47:14.844 file.go:Get:69 → cache is nil, returning empty value for key: toggle_cache
[TRACE] 23:47:14.844 terminal.go:GOOS() - 122ns
[DEBUG] 23:47:14.844 segment.go:isToggled:234 → no toggles found
[TRACE] 23:47:14.844 terminal.go:GOOS() - 22ns
[TRACE] 23:47:14.844 terminal.go:GOOS() - 22ns
[DEBUG] 23:47:14.844 terminal.go:Getenv:77 → NO DATA
[TRACE] 23:47:14.844 terminal.go:GOOS() - 22ns
[TRACE] 23:47:14.844 terminal.go:Getenv(SSH_CONNECTION) - 4.585µs
[TRACE] 23:47:14.844 terminal.go:GOOS() - 22ns
[DEBUG] 23:47:14.844 terminal.go:Getenv:77 → NO DATA
[TRACE] 23:47:14.844 terminal.go:Flags() - 77ns
[TRACE] 23:47:14.844 terminal.go:Getenv(SSH_CLIENT) - 2.358µs
[DEBUG] 23:47:14.844 file.go:Get:80 → getting cache key: environment_platform with value: ubuntu
[DEBUG] 23:47:14.844 terminal_unix.go:Platform:92 → ubuntu
[DEBUG] 23:47:14.844 properties.go:GetStringArray:52 → cycle: []
[DEBUG] 23:47:14.844 properties.go:GetString:28 → folder_separator_template:
[DEBUG] 23:47:14.844 properties.go:GetString:28 → folder_separator_icon: /
[DEBUG] 23:47:14.844 properties.go:GetBool:22 → cycle_folder_separator: false
[DEBUG] 23:47:14.844 properties.go:GetString:28 → folder_format: %s
[DEBUG] 23:47:14.844 properties.go:GetString:28 → edge_format: %s
[DEBUG] 23:47:14.844 properties.go:GetString:28 → left_format: %s
[DEBUG] 23:47:14.844 properties.go:GetString:28 → right_format: %s
[TRACE] 23:47:14.844 terminal.go:Flags() - 33ns
[TRACE] 23:47:14.844 terminal.go:GOOS() - 22ns
[TRACE] 23:47:14.844 terminal.go:StackCount() - 44ns
[TRACE] 23:47:14.844 terminal_unix.go:DirIsWritable(/home/omp-test) - 7.858µs
[DEBUG] 23:47:14.846 terminal.go:RunCommand:293 → NO DATA
[TRACE] 23:47:14.846 terminal.go:RunCommand(who am i) - 2.248041ms
[DEBUG] 23:47:14.847 text.go:patchTemplate:125 →  {{ if .Data.SSHSession }} {{ end }}{{ .UserName }}
[TRACE] 23:47:14.847 text.go:Render( {{ if .SSHSession }} {{ end }}{{ .UserName }} ) - 360.078µs
[TRACE] 23:47:14.847 text.go:Render(diamond) - 77ns
[DEBUG] 23:47:14.847 text.go:patchTemplate:125 →   {{ path .Data.Path .Data.Location }}
[TRACE] 23:47:14.847 text.go:Render(  {{ path .Path .Location }} ) - 62.238µs
[TRACE] 23:47:14.847 text.go:Render(powerline) - 66ns
[TRACE] 23:47:14.847 text.go:Render(powerline) - 44ns
[TRACE] 23:47:14.847 text.go:Render(powerline) - 22ns
[TRACE] 23:47:14.847 text.go:Render( CONFIG ERROR ) - 33ns
[TRACE] 23:47:14.847 text.go:Render(p:red) - 33ns
[TRACE] 23:47:14.847 text.go:Render(diamond) - 165ns
[TRACE] 23:47:14.847 text.go:Render(p:red) - 33ns
[TRACE] 23:47:14.847 terminal.go:Shell() - 99ns
[TRACE] 23:47:14.847 terminal.go:Shell() - 22ns
[TRACE] 23:47:14.847 terminal.go:Flags() - 298ns
[DEBUG] 23:47:14.847 home.go:12 → /home/omp-test
[TRACE] 23:47:14.847 terminal.go:GOOS() - 33ns
[DEBUG] 23:47:14.847 segment.go:Execute:113 → segment: Time
[DEBUG] 23:47:14.847 file.go:Get:69 → cache is nil, returning empty value for key: toggle_cache
[DEBUG] 23:47:14.847 segment.go:isToggled:234 → no toggles found
[DEBUG] 23:47:14.847 properties.go:GetString:28 → time_format: 15:04:05
[TRACE] 23:47:14.847 terminal.go:Flags() - 187ns
[DEBUG] 23:47:14.847 home.go:12 → /home/omp-test
[TRACE] 23:47:14.847 terminal.go:GOOS() - 22ns
[DEBUG] 23:47:14.847 segment.go:Execute:113 → segment: Node
[DEBUG] 23:47:14.847 file.go:Get:69 → cache is nil, returning empty value for key: toggle_cache
[DEBUG] 23:47:14.847 segment.go:isToggled:234 → no toggles found
[DEBUG] 23:47:14.847 home.go:12 → /home/omp-test
[TRACE] 23:47:14.847 terminal.go:GOOS() - 44ns
[DEBUG] 23:47:14.847 segment.go:Execute:113 → segment: Python
[DEBUG] 23:47:14.847 file.go:Get:69 → cache is nil, returning empty value for key: toggle_cache
[TRACE] 23:47:14.847 terminal.go:Flags() - 331ns
[DEBUG] 23:47:14.847 segment.go:isToggled:234 → no toggles found
[DEBUG] 23:47:14.847 home.go:12 → /home/omp-test
[TRACE] 23:47:14.847 terminal.go:GOOS() - 33ns
[DEBUG] 23:47:14.847 properties.go:GetStringArray:52 → extensions: [*.js *.ts package.json .nvmrc pnpm-workspace.yaml .pnpmfile.cjs .vue]
[DEBUG] 23:47:14.847 file.go:Get:69 → cache is nil, returning empty value for key: toggle_cache
[DEBUG] 23:47:14.847 segment.go:isToggled:234 → no toggles found
[DEBUG] 23:47:14.847 properties.go:GetString:28 → display_mode: files
[DEBUG] 23:47:14.847 properties.go:GetStringArray:52 → folders: []
[DEBUG] 23:47:14.847 properties.go:GetBool:22 → home_enabled: false
[DEBUG] 23:47:14.847 properties.go:GetStringArray:52 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 23:47:14.847 home.go:12 → /home/omp-test
[TRACE] 23:47:14.847 terminal.go:GOOS() - 22ns
[DEBUG] 23:47:14.847 properties.go:GetStringArray:52 → folders: [.venv venv virtualenv venv-win pyenv-win]
[DEBUG] 23:47:14.847 segment.go:Execute:113 → segment: Shell
[DEBUG] 23:47:14.847 properties.go:GetBool:22 → home_enabled: false
[DEBUG] 23:47:14.847 file.go:Get:69 → cache is nil, returning empty value for key: toggle_cache
[DEBUG] 23:47:14.847 home.go:12 → /home/omp-test
[DEBUG] 23:47:14.847 segment.go:isToggled:234 → no toggles found
[DEBUG] 23:47:14.847 properties.go:GetKeyValueMap:46 → mapped_shell_names: map[]
[TRACE] 23:47:14.847 terminal.go:Shell() - 55ns
[TRACE] 23:47:14.847 text.go:Render(plain) - 286ns
[TRACE] 23:47:14.847 terminal.go:Flags() - 22ns
[TRACE] 23:47:14.847 text.go:Render(plain) - 44ns
[TRACE] 23:47:14.847 text.go:Render(plain) - 55ns
[DEBUG] 23:47:14.847 text.go:patchTemplate:125 → in <p:blue><b>{{ .Data.Name }}</b></>
[TRACE] 23:47:14.849 text.go:Render(in <p:blue><b>{{ .Name }}</b></> ) - 1.867286ms
[TRACE] 23:47:14.849 text.go:Render(plain) - 66ns
[DEBUG] 23:47:14.849 text.go:patchTemplate:125 → at <p:blue><b>{{ .Data.CurrentDate | date "15:04:05" }}</b></>
[TRACE] 23:47:14.849 text.go:Render(at <p:blue><b>{{ .CurrentDate | date "15:04:05" }}</b></>) - 39.324µs
[TRACE] 23:47:14.849 text.go:Render(plain) - 44ns
[TRACE] 23:47:14.849 terminal.go:Shell() - 88ns
[TRACE] 23:47:14.849 terminal.go:Shell() - 22ns
[TRACE] 23:47:14.849 terminal.go:Flags() - 33ns
[TRACE] 23:47:14.849 terminal.go:Shell() - 22ns
[DEBUG] 23:47:14.851 terminal_unix.go:TerminalWidth:85 → terminal width: 120
[TRACE] 23:47:14.851 terminal_unix.go:TerminalWidth() - 1.332643ms
[DEBUG] 23:47:14.851 home.go:12 → /home/omp-test
[TRACE] 23:47:14.851 path.go:Path() - 21.767µs
[TRACE] 23:47:14.851 terminal.go:Flags() - 33ns
```

--- a/src/config/default.go
+++ b/src/config/default.go
@@ -1,6 +1,8 @@
 package config
 
 import (
+	"github.com/jandedobbeleer/oh-my-posh/src/cache"
+	"github.com/jandedobbeleer/oh-my-posh/src/cli/upgrade"
 	"github.com/jandedobbeleer/oh-my-posh/src/color"
 	"github.com/jandedobbeleer/oh-my-posh/src/properties"
 	"github.com/jandedobbeleer/oh-my-posh/src/segments"
@@ -193,6 +195,12 @@ func Default(warning bool) *Config {
 				},
 				Tips: []string{"az"},
 			},
+		},
+		Upgrade: &upgrade.Config{
+			Source:        upgrade.CDN,
+			DisplayNotice: true,
+			Auto:          true,
+			Interval:      cache.ONEWEEK,
 		},
 	}
 

